src/nifs/tests.rs

```diff
@@ -0,0 +1,695 @@
+
+    #[derive(Debug, Clone)]
+    struct FiboConfig {
+        a: Column<Advice>,
+        b: Column<Advice>,
+        selector: Selector,
+        instance: Column<Instance>,
+    }
+
+    struct FiboChip<F: PrimeField> {
+        config: FiboConfig,
+        _marker: PhantomData<F>,
+    }
+
+    impl<F: PrimeField> FiboChip<F> {
+        fn construct(config: FiboConfig) -> Self {
+            Self {
+                config,
+                _marker: PhantomData,
+            }
+        }
+
+        fn configure(
+            meta: &mut ConstraintSystem<F>,
+            advice: [Column<Advice>; 2],
+            selector: Selector,
+            instance: Column<Instance>,
+        ) -> FiboConfig {
+            let a = advice[0];
+            let b = advice[1];
+
+            meta.enable_equality(b);
+            meta.enable_equality(instance);
+
+            meta.create_gate("fibo-block", |meta| {
+                let s = meta.query_selector(selector);
+                let a1 = meta.query_advice(a, Rotation::prev());
+                let b1 = meta.query_advice(b, Rotation::prev());
+                let a2 = meta.query_advice(a, Rotation::cur());
+                let b2 = meta.query_advice(b, Rotation::cur());
+                vec![
+                    s.clone() * (a2.clone() - b1.clone() - a1),
+                    s * (b2 - a2 - b1),
+                ]
+            });
+
+            FiboConfig {
+                a,
+                b,
+                selector,
+                instance,
+            }
+        }
+
+        fn load(
+            &self,
+            mut layouter: impl Layouter<F>,
+            a: F,
+            b: F,
+            nrows: usize,
+        ) -> Result<(Number<F>, Number<F>), Error> {
+            layouter.assign_region(
+                || "entire block",
+                |mut region| {
+                    // assign first row
+                    let mut a = region
+                        .assign_advice(|| "a", self.config.a, 0, || Value::known(a))
+                        .map(Number)?;
+
+                    let mut b = region
+                        .assign_advice(|| "b", self.config.b, 0, || Value::known(b))
+                        .map(Number)?;
+                    // println!("[0] a = {:?} b = {:?}", a.0, b.0);
+
+                    for idx in 1..nrows {
+                        self.config.selector.enable(&mut region, idx)?;
+                        let a2 = a.0.value().and_then(|a| b.0.value().map(|b| *a + *b));
+                        let b2 = a2.and_then(|a| b.0.value().map(|b| a + *b));
+
+                        a = region
+                            .assign_advice(|| "a", self.config.a, idx, || a2)
+                            .map(Number)?;
+
+                        b = region
+                            .assign_advice(|| "b", self.config.b, idx, || b2)
+                            .map(Number)?;
+                        // println!("[{}] a = {:?} b = {:?}", idx, a.0, b.0);
```

Comment:
use `#[test_log::test]` + `log::debug` for staff like that. And with `RUST_LOG=debug` you always recieve all needed messages without comment\uncomment

Reply:
good point. This is actually directly copied from the online example (see the http link in very beginning) so I forgot to remove it.